src/core/thread/mle_router.cpp

```diff
@@ -2035,6 +2035,7 @@ ThreadError MleRouter::HandleChildUpdateRequest(const Message &aMessage, const I
     }
 
     child->mLastHeard = Timer::GetNow();
+    child->mAddSrcMatchEntryShort = true;
```

Comment:
I agree that the source match table needs to start with an extended address then generally moves to the short address.  By switching to using the short on receiving the first Data Request with a short address, we know that the child has begun polling with its short address.

The problem I'm trying to address is that I could not find anything in the Thread Spec regarding the Source Address mode for MAC Data Request messages.  As a result, a device could utilize either short or extended address modes as it wishes.